templates/zone-master.erb

```diff
@@ -1,6 +1,6 @@
 <%-
-if @is_dynamic and @allow_update.empty?
-  raise(Puppet::ParseError, "allow_update is empty for dynamic zone '#{name}'")
+if @is_dynamic and (@allow_update.empty? and @allow_update_cidr.empty?)
+  raise(Puppet::ParseError, "Both allow_update and allow_update_cidr are empty for dynamic zone '#{name}'")
```

Comment:
Well, IMO it's a perfectly normal scenario: I can both accept updates from a secure network via CIDR validation and from an unsecure one with the secret key.
I don't understand what you mean to use `is_a?(Array)`, do you mean instead of `.empty?` ?